flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -1528,6 +1532,51 @@ IntrinsicLibrary::genPresent(mlir::Type,
                                           fir::getBase(args[0]));
 }
 
+// SCAN 
+fir::ExtendedValue
+IntrinsicLibrary::genScan(mlir::Type resultType,
+                          llvm::ArrayRef<fir::ExtendedValue> args) {
+
+  assert(args.size() >= 2);
+
+  // Handle required string argument
+  auto string = builder.createBox(loc, args[0]);
+  // Handle required set argument
+  auto set = builder.createBox(loc, args[1]);
+
+  // Handle optional argument, back
+  auto back = fir::isUnboxedValue(args[2])
```

Comment:
You can use `!isAbsent(args[2])` that I added in #703. (maybe reverse the logic to avoid the `!`). Your code is correct, but I think using `isAbsent` will be more expressive.

Reply:
Some would argue that negated logic tests are less readable.

Reply:
It should be pointed out that `!isAbsent()` is not identical to `isUnboxedValue()`. The former returns true for any non-null extended value regardless of type. The latter only returns true for a non-null ssa-value that is not boxed, not a character, and not an array.